### ISSUE TITLE
"dotnet msbuild /t:package" doesn't work

### ISSUE THREAD

**Issue by Aleksanderis** [Azure/service-fabric-issues#258]
Not sure if it was already discussed - I found some similar topics, but still not exactly same.

I faced with some real challenges mixing Service Fabric and .NET Core solution. 
Through Visual Studio and using `C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\msbuild.exe myproj.csproj /t:package` - everything builds and works fine. However it's not possible to build it on other machines with no VS2017 installed (e.g. CI server), eventhough there is latest .NET Core SDK present.

So the issue is:
 `C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\msbuild.exe myproj.csproj /t:package` - works
But:
 `dotnet msbuild myproj.csproj /t:package` - doesn't work and throws:

> c:\Projects\MyProj\Source\packages\Microsoft.VisualStudio.Azure.Fabric.MSBuild.1.5.0\build\Microsoft.VisualStudio.Azure.Fabric.Application.targets(77,5): error MSB4018: The "ValidatePackageSettings" task failed unexpectedly.
> c:\Projects\MyProj\Source\packages\Microsoft.VisualStudio.Azure.Fabric.MSBuild.1.5.0\build\Microsoft.VisualStudio.Azure.Fabric.Application.targets(77,5): error MSB4018: System.IO.FileNotFoundException: Could not load file or assembly 'System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'. The system cannot find the file specified. [c:\Projects\MyProj\Source\MyProj.sfproj]

There is an older Visual Studio 2015 present with msbuild 14.0 on CI server, but it also doesn't work, because it fails to parse .NET Core *.csproj files.

Is it really the only option to install VS2017 on CI server? My first impression was that `dotnet msbuild` is same as `msbuild.exe`, and also in Microsoft docs (https://docs.microsoft.com/en-us/visualstudio/msbuild/what-s-new-in-msbuild-15-0) it says - `MSBuild is now available as part of the .NET Core SDK and can build .NET Core projects on Windows, macOS, and Linux.`

Am I doing something wrong, or is it a known issue? Can it be workarounded somehow?

**Comment by cwe1ss** [comment]
I'm having the same issue. I'd love to see this "just work" with `dotnet publish` on the sfproj project.

**Comment by MikkelHegn** [comment]
@ravipal

**Comment by ravipal** [comment]
We support dotnet core services, but the application project (*.sfproj) is still '.Net Framework', so dotnet msbuild is not supported.

**Comment by Aleksanderis** [comment]
@ravipal, yes, I understand it, but the problem that from one side `dotnet msbuild` is not supported, from another (from .NET Core services) - `"classic" msbuild` is not supported (it cannot build .NET Core). So it's a lose-lose situation.
Both technologies are part of Microsoft stack, that's why it's so strange. And it's just a slight step aside from happy-path, when it stops working.
Do you know if there are any close plans to fix this? E.g. adding support for *.sfproj in new format, to make it fully compatible with .NET Core projects.

**Comment by klettier** [comment]
I'm sad that one year later this is still an issue :(
I ended up doing
* Restore each dotnet core projects separately (dotnet restore)
* Restore nuget packages on solution (nuget restore)
* Package SF project file (msbuild /t:package)

This is for the moment the quickest way to build an SF solution mixing dotnet core projects and SF project.

**Comment by devlie** [comment]
We are doing something similar where we do:
1. dotnet restore on .NET Core csproj
2. dotnet build on .sln
3. nuget restore on .sln
4. msbuild /t:package on sfproj

But we found that msbuild /t:package always execute build instead of just packaging. Is there a way to skip build when packaging?

**Comment by abatishchev** [comment]
Why not `dotnet pack` then?

**Comment by klettier** [comment]
@devlie you should be able to remove ```dotnet build on sln``` and use ``` msbuild /t:package``` to build and produce package

**Comment by devlie** [comment]
@abatishchev , as far as I know, `dotnet pack` doesn't support packaging SF package.

@klettier I could have, but `msbuild /t:package` only builds SF app's dependencies (my service code), and I have my client code csproj in the same solution which is not getting built. Can be solved with custom steps, was just wondering if there's anything more convenient 🤔

**Comment by abatishchev** [comment]
Right! It doesn't. Sorry, got confused.

We're using "raw" nuspec to package SF: pkg -> sfpkg -> nupkg.
